packages/content-conformance/src/rules/ensure-html-comments-on-own-line.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable import/no-anonymous-default-export */
+/** @type {import('../types.js').ConformanceRuleBase} */
+export default {
+  type: 'content',
+  id: 'ensure-html-comments-on-own-line',
+  description: 'Ensures that HTML comments are on their own line.',
+  executor: {
+    async contentFile(file, context) {
+      file.visit('jsx', (node) => {
+        // is HTML comment or JSX tag with HTML comment
+        if (node.value.includes('<!--')) {
+          // if child of another node
+          if (node.position.start.column != 1) {
+            context.report(
+              'Detected an HTML comment that is likely nested. You may be using this in a bullet point or other markdown primitive, which is not permitted. Please ensure HTML comments are on their own line, without any indentation or nesting.',
+              file,
+              node
+            )
+          }
+
+          // if leading whitespace
+          if (node.value.match(/^\s/i)) {
+            context.report(
+              'Detected an HTML comment that is indented. This is not permitted. Please ensure HTML comments are on their own line, without any indentation or nesting.',
+              file,
+              node
+            )
+          }
+
+          // likely nested in JSX tag
+          if (!node.value.startsWith('<!')) {
+            context.report(
+              'Detected an HTML comment that is nested within a JSX Tag. This is not permitted. Please ensure HTML comments are on their own line, without any indentation or nesting.',
+              file,
+              node
+            )
+          }
+        }
+      })
+
+      // check for accidental 4-space indentation (codeblock)
+      file.visit('code', (node) => {
+        if (node.value.includes('<!--')) {
+          context.report(
+            'Detected an HTML comment in a code block. This is not permitted. Please ensure HTML comments are on their own line, without any indentation or nesting.',
+            file,
+            node
+          )
+        }
+      })
```

Comment:
I'm a little nervous about this one, what if someone has a legitimate HTML comment as part of a codeblock? It seems unlikely, but it's technically not invalid.

Reply:
Yea, I thought about this case and concluded in my head that we realistically won't encounter this situation.

Technically we could add a check if `node.lang === 'html'` (and make it a requirement that `html` codeblocks be marked with `html`) but this seems excessive 🤔